prow/tide/status.go

```diff
@@ -288,10 +290,16 @@ func (sc *statusController) setStatuses(all []PullRequest, pool map[string]PullR
 			log.WithError(err).Error("Getting head commit status contexts, skipping...")
 			return
 		}
-		cr, err := sc.config().GetTideContextPolicy(
-			string(pr.Repository.Owner.Login),
-			string(pr.Repository.Name),
-			string(pr.BaseRef.Name))
+
+		org := string(pr.Repository.Owner.Login)
+		repo := string(pr.Repository.Name)
+		branch := string(pr.BaseRef.Name)
+		headSHA := string(pr.HeadRefOID)
+		baseSHAGetter := func() (string, error) {
+			return sc.ghc.GetRef(org, repo, "heads/"+branch)
```

Comment:
Well want to cache the sha so that we only look it up once per pool instead of once per PR. This doesn't block merge though since we won't actually call the baseSHAGetter at all right now.